ui/components/MesherySettings.js

```diff
@@ -134,205 +134,147 @@ TabContainer.propTypes = { children : PropTypes.node.isRequired, };
+       JSON.stringify(props.meshAdapters) !== JSON.stringify(meshAdapters)
+    ) {
+
+      k8sconfig (props.k8sconfig)
+      meshAdapters (props.meshAdapters)
+      grafana (props.grafana)
+      prometheus ( props.prometheus)
 
-  static getDerivedStateFromProps(props, state) {
-    let st = {};
-    if (JSON.stringify(props.k8sconfig) !== JSON.stringify(state.k8sconfig)
-      || JSON.stringify(props.meshAdapters) !== JSON.stringify(state.meshAdapters)) {
-      st = {
-        k8sconfig : props.k8sconfig,
-        meshAdapters : props.meshAdapters,
-        grafana : props.grafana,
-        prometheus : props.prometheus,
-      };
     }
-    const compare = (arr1, arr2) => arr1.every((val, ind) => val === arr2[ind])
+    const compare = (arr1, arr2) => arr1.every((val, ind) => val === arr2[ind]);
 
-    if (props.telemetryUrls.grafana.length !== state.scannedGrafana.length || !(compare(props.telemetryUrls.grafana, state.scannedGrafana))) {
-      st.scannedGrafana = props.telemetryUrls.grafana
+    if (
+      props.telemetryUrls.grafana.length !== scannedGrafana.length ||
+       !compare(props.telemetryUrls.grafana, scannedGrafana)
+    ) {
+      setScannedGrafana(props.telemetryUrls.grafana);
     }
 
-    if (props.telemetryUrls.prometheus.length !== state.scannedPrometheus.length || !(compare(props.telemetryUrls.prometheus, state.scannedPrometheus))) {
-      st.scannedPrometheus = props.telemetryUrls.prometheus
+    if (
+      props.telemetryUrls.prometheus.length !==
+         scannedPrometheus.length ||
+       !compare(props.telemetryUrls.prometheus, scannedPrometheus)
+    ) {
+      setScannedPrometheus(props.telemetryUrls.prometheus);
     }
-    return st;
-  }
+  }, [props.k8sconfig, props.meshAdapters, props.grafana, props.prometheus, props.telemetryUrls]);
 
-  async componentDidMount() {
+  const fetchData = async () => {
     try {
       const modelsResponse = await getModelsDetail();
       const componentsResponse = await getComponentsDetail();
       const relationshipsResponse = await getRelationshipsDetail();
 
-      const modelsCount = modelsResponse.total_count;
-      const componentsCount = componentsResponse.total_count;
-      const relationshipsCount = relationshipsResponse.total_count;
-
-      this.setState({
-        modelsCount,
-        componentsCount,
-        relationshipsCount
-      });
+      setModelsCount(modelsResponse.total_count);
+      setComponentsCount(componentsResponse.total_count);
+      setRelationshipsCount(relationshipsResponse.total_count);
     } catch (error) {
       console.error(error);
     }
-  }
-
+  };
+  useEffect(() => {
+    fetchData();
+    return () => {
+      setModelsCount(0);
+      setComponentsCount(0);
+      setRelationshipsCount(0);
+    };
+  }, []);
 
-  handleError = (msg) => (error) => {
-    this.props.updateProgress({ showProgress : false });
-    const notify = this.props.notify;
+  // eslint-disable-next-line no-unused-vars
```

Comment:
dont disable lint rules

Reply:
Actually @aabidsofi19 we are not using this handleError function anywere . so that why i am getting this error 
Error: 'handleError' is assigned a value but never used . when restarting the server . that's why i disable lint rules for this function only . can we remove this function.

Reply:
Yes, you can remove the function if it's not being used anywhere.  Or you could just comment that out as well marking as not being used.

Reply:
> Yes, you can remove the function if it's not being used anywhere. Or you could just comment that out as well marking as not being used.

ok

Reply:
@Rajdip019 @aabidsofi19 
I have commented  the Unused function as marking as it is not in use .